src/rtctools/optimization/seed_mixin.py

```diff
@@ -0,0 +1,129 @@
+    """
+
+    def __init__(self, **kwargs):
+        self._seed_timeseries: Path = None
+        super().__init__(**kwargs)
+
+    def _use_seed_timeseries(self):
+        """Return True if a seed from a timeseries is used for the current run."""
+        if self._seed_timeseries is None:
+            return False
+        if isinstance(self, GoalProgrammingMixin):
+            if not self._gp_first_run:
+                return False
+        if isinstance(self, HomotopyMixin):
+            theta_name = self.homotopy_options()["homotopy_parameter"]
+            theta = self.parameters(ensemble_member=0)[theta_name]
+            theta_start = self.homotopy_options()["theta_start"]
+            if theta > theta_start:
+                return False
+        return True
+
+    def seed_options(self) -> dict:
+        """
+        Get the seed options.
+
+        +---------------------------+-----------+---------------+
+        | Option                    | Type      | Default value |
+        +===========================+===========+===============+
+        | ``seed_timeseries``       | ``Path``  | ``None``      |
+        +---------------------------+-----------+---------------+
+        | ``is_binary``             | ``bool``  | ``False``     |
+        +---------------------------+-----------+---------------+
+        | ``validate_timeseries``   | ``bool``  | ``False``     |
+        +---------------------------+-----------+---------------+
+        | ``times_are_equidistant`` | ``bool``  | ``False``     |
+        +---------------------------+-----------+---------------+
+        | ``fallback``              | ``bool``  | ``False``     |
+        +---------------------------+-----------+---------------+
+
+        - seed_timeseries:
+            Path to the file containing the seed.
+            If None, the default seed is used.
+        - is_binary:
+            Indicates if the file is in binary format.
+        - validate_timeseries:
+            Indicates if the timeseries should be validated.
+        - times_are_equidistant:
+            Indicates if the time stamps are equidistant.
+        - fallback:
+            Indicates if the solver should fallback to the default seed
+            if it fails for the given seed.
+        """
+        return {
+            "seed_timeseries": None,
+            "is_binary": False,
+            "validate_timeseries": False,
+            "times_are_equidistant": False,
+            "fallback": False,
+        }
+
+    def seed_timeseries(self) -> Path:
+        """Get the path of the seed timeseries."""
+        return self._seed_timeseries
+
+    def seed(self, ensemble_member):
+        if not self._use_seed_timeseries():
+            return super().seed(ensemble_member)
+        seed: dict = super().seed(ensemble_member).copy()  # Copy to prevent updating cached seeds.
+        if isinstance(self, CSVMixin):
+            times, values_dict = get_timeseries_from_csv(self._seed_timeseries)
```

Comment:
This code introduces a dependency between 'SeedMixin' and 'CSVMixin', which implicitly assumes that the 'SeedMixin' would only be used in combination with either 'CSVMixin' or 'PIMixin'. As a result, it would not work well with other, 3rd party IO mixins.

Reply:
Yes, it doesn't extend nicely if a user would use another file type. I'll try to find another approach.

Reply:
The user is now responsible for implementing the method seed_from_file. Using the get_timeseries functions, this requires only a few lines.